pkg/apis/core/validation/deployitem.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/gardener/landscaper/pkg/apis/core"
+)
+
+// ValidateDeployItem validates a DeployItem
+func ValidateDeployItem(di *core.DeployItem) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, ValidateDeployItemSpec(field.NewPath("spec"), di.Spec)...)
+
+	return allErrs
+}
+
+// ValidateDeployItemSpec validates a DeployItem spec
+func ValidateDeployItemSpec(fldPath *field.Path, diSpec core.DeployItemSpec) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if diSpec.Type == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("type"), "type is required"))
```

Comment:
Sometimes we use `must not be emtpy` and somtimes `is required`.
I think we should stick with one error message